File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
            stack<char> st;
            string result = "";

            for (int i = 0; i < s.length(); i++) {

                if (s[i] == '(') {
                    if (!st.empty()) {
                        result += s[i];
                    }
                    st.push(s[i]);

                } else if (s[i] == ')') {
                    st.pop();
                    if (!st.empty()) {
                        result += s[i];
                    }

                }
            }
            return result;
        // string ans = "";
        // int depth = 0;

        // for (int i = 0; i < s.length(); i++) {
        //     if (s[i] == '(') {
        //         depth++;
        //         if (depth > 1) {
        //             ans += s[i];
        //         }
                
        //     } else if (s[i] == ')') {

        //         if (depth > 1) {
        //             ans += s[i];
        //         }
        //         depth--;
        //     }
        // }
        // return ans;
    }
};

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        int depth = 0;
        StringBuilder ans = new StringBuilder(); // ✅ Use StringBuilder instead of char[] (which isn't resizable)

        for (char ch : s.toCharArray()) {
            
            if (ch == '(') {
                depth++;
                if (depth > 1) {
                    ans.append(ch);  // ✅ use append for adding characters
                }
            }

            if (ch == ')') {
                if (depth > 1) {
                    ans.append(ch);  // ✅ use append for adding characters
                }
                depth--;
            }
        }

        return ans.toString(); // ✅ convert StringBuilder to String
    }
}
